### PR TITLE
update Tags format

### PR DESCRIPTION
**Pull request by chunhtai**

### CODE REVIEW

**Comment by chunhtai** [comment]
For the incoming ci format change https://github.com/flutter/flutter/pull/85565/files

File: app_dart/lib/src/model/proto/internal/scheduler.proto [hunk 1]
```diff
@@ -57,7 +57,22 @@ message Target {
     // Name of the LUCI recipe to use for the builder.
     optional string recipe = 13;
     // Additional tag for the target.
-    repeated string tags = 14;
+    optional Tags tags = 14;
```
Comment (CaseyHillers) [inline app_dart/lib/src/model/proto/internal/scheduler.proto:60]: It seems simpler to do

```
repeated Tags tags = 14;
```

And have tags be an enum

Comment (chunhtai) [reply app_dart/lib/src/model/proto/internal/scheduler.proto:60]: The plan is tags can be extended to have complex value
for example,

   tags:
        shard: true
        platform: ios
        duration: 1000

Comment (CaseyHillers) [reply app_dart/lib/src/model/proto/internal/scheduler.proto:60]: Is there an issue or a doc I can look at? It sounds like we're working on similar functionality (https://github.com/flutter/flutter/issues/84998, go/flutter-automatic-builders)

Comment (chunhtai) [reply app_dart/lib/src/model/proto/internal/scheduler.proto:60]: This is a part of
https://docs.google.com/document/d/1DRw_J3zjuVpiVRN3iRRCjeI0RGoLrB6zOOTtj4h-hHI/edit#heading=h.4qsx9pjitde1

the problem i am try to solve here is that we need to find testowner and file pr to mark test flaky. Depends on the type of the test, the way to find test owner and file pr will be different. That is why I want to add different tag to those tests, so the script can pick it up. While I was making the change, @keyonghan suggest we make the tags to be a complex object so people extend it in the future.

Comment (CaseyHillers) [reply app_dart/lib/src/model/proto/internal/scheduler.proto:60]: Thanks for the context. I believe we can pull this information without needing to modify `.ci.yaml`.

`platform` can be retrieved by the first word of the target.name: `Linux_android`, `Linux`, etc
`shard` is determined by if there is `_#` or `_last` at the end of the target name
`duration` is `target.timeout` (I'm working on a PR to mass enable these)
The test types are harder to translate, but if we could simplify to on-device tests + vm tests we could reuse platform.

The word `tag` is misleading as this file is intended to be used by `LUCI` where there's a different connotation for tag.

Comment (chunhtai) [reply app_dart/lib/src/model/proto/internal/scheduler.proto:60]: > shard is determined by if there is _# or _last at the end of the target name

I think some do not follow that, e.g. framework_tests_libraries, framework_tests_misc. I also need to know whether it is framework test, devicelab test.. etc

platform and duration are just examples for future extension that people can create complex key value pair

> The word tag is misleading as this file is intended to be used by LUCI where there's a different connotation for tag.

If so, we can consider a different name

File: app_dart/lib/src/model/proto/internal/scheduler.proto [hunk 2]
```diff
@@ -57,7 +57,22 @@ message Target {
...
+}
+
+// Tags for a specific infrastructure task
+// Next ID: 6
+message Tags {
```
Comment (CaseyHillers) [inline app_dart/lib/src/model/proto/internal/scheduler.proto:65]: Based on [builds_service.proto](https://source.chromium.org/chromium/infra/infra/+/main:go/src/go.chromium.org/luci/buildbucket/proto/builds_service.proto;l=280;drc=adbd1e588d722858df732b1ecb23e09c00ab1b54), can we make this a StringPair as well?

Comment (chunhtai) [reply app_dart/lib/src/model/proto/internal/scheduler.proto:65]: sounds good

Comment (chunhtai) [reply app_dart/lib/src/model/proto/internal/scheduler.proto:65]: it looks like StringPair is a custom structure defined in https://source.chromium.org/chromium/infra/infra/+/main:luci/appengine/swarming/proto/api/swarming.proto?q=StringPair&ss=chromium%2Finfra%2Finfra

I will need to duplicate the code I assume?

Comment (CaseyHillers) [reply app_dart/lib/src/model/proto/internal/scheduler.proto:65]: Yeah, easier to copy it here since it's relatively simple

**Comment by chunhtai** [comment]
Hi Casey, I updated the format PTAL :)

File: app_dart/lib/src/model/proto/internal/scheduler.proto [hunk 3]
```diff
@@ -57,7 +57,17 @@ message Target {
     // Name of the LUCI recipe to use for the builder.
     optional string recipe = 13;
     // Additional tag for the target.
-    repeated string tags = 14;
+    repeated StringPair tags = 14;
```
Comment (CaseyHillers) [inline app_dart/lib/src/model/proto/internal/scheduler.proto:60]: doc?

File: app_dart/lib/src/model/proto/internal/scheduler.proto [hunk 4]
```diff
@@ -57,7 +57,17 @@ message Target {
...
+}
+
+// A string key-value pair. Typically used for tagging.
+//
+// This structure need to be in sync with the StringPair in LUCI infra.
+// See the `go/src/go.chromium.org/luci/buildbucket/proto/common.proto` in the
+// https://source.chromium.org
+message StringPair {
```
Comment (CaseyHillers) [inline app_dart/lib/src/model/proto/internal/scheduler.proto:68]: nit: next id?

File: app_dart/lib/src/model/proto/internal/scheduler.proto [hunk 5]
```diff
@@ -57,7 +57,17 @@ message Target {
     // Name of the LUCI recipe to use for the builder.
     optional string recipe = 13;
     // Additional tag for the target.
-    repeated string tags = 14;
```
Comment (CaseyHillers) [inline app_dart/lib/src/model/proto/internal/scheduler.proto:60]: Can we link to the LUCI docs on what a tag is?

**Review by CaseyHillers — APPROVED** [review]
LGTM once the docs comments are addressed

File: app_dart/lib/src/model/proto/internal/scheduler.proto (hunk 4 above)
Comment (chunhtai) [reply app_dart/lib/src/model/proto/internal/scheduler.proto:68]: This structure should not be changed, do we still need to document next id?

Comment (CaseyHillers) [reply app_dart/lib/src/model/proto/internal/scheduler.proto:68]: Good point, SGTM

File: app_dart/lib/src/model/proto/internal/scheduler.proto (hunk 5 above)
Comment (chunhtai) [reply app_dart/lib/src/model/proto/internal/scheduler.proto:60]: I am hesitate to do that now seems these tags are not feed into the luci yet. Should we wait for that once we actually feed this tags into luci?

**Pull request merged**